### PR TITLE
[istio] Istiod chore pack #1

### PR DESCRIPTION
**Pull request by apolovov**
## Description
* Anti-affinity for istiod pods in HA installations.
* `proxyConfig.holdApplicationUntilProxyStarts` global flag to guarantee that istio sidecar starts before application container.
* `enableHTTP10` flag to allow HTTP/1.0 requests handling.

## Why do we need it, and what problem does it solve?
* It could be traffic issues in large installations when istiod pods are being scheduled to a single node.
* There were issues with applications that cannot handle connection errors at startup.
* There are issues with legacy software that uses HTTP/1.0 requests by default and it isn't that simple to fix it.

## Changelog entries

```changes
section: istio
type: chore
summary: Anti-affinity for istiod pods in HA installations. `proxyConfig.holdApplicationUntilProxyStarts` global flag to guarantee that istio sidecar starts before application container. `enableHTTP10` flag to allow HTTP/1.0 requests handling.

```

### CODE REVIEW

File: werf.yaml [hunk 1]
```diff
@@ -333,7 +333,7 @@ ansible:
   - name: "Install GolangCI linter"
     shell: |
       curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
-        | sh -s -- -b $(go env GOPATH)/bin v1.42.0
+        | sh -s -- -b $(go env GOPATH)/bin v1.46.2
```
Comment (apolovov) [inline werf.yaml:336]: v1.42.0 disappeared.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @apolovov. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @apolovov. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @apolovov. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @apolovov. I've started the workflow for label 'deploy/web/test'...

**Review by yalosev — APPROVED** [review]

**Review by yalosev — DISMISSED** [review]

File: ee/modules/110-istio/openapi/doc-ru-config-values.yaml [hunk 2]
```diff
@@ -132,6 +135,15 @@ properties:
         description: Цепочка сертификатов в формате PEM на случай  если `cert` — промежуточный сертификат.
       root:
         description: Корневой сертификат в формате PEM на  случай если `cert` — промежуточный сертификат.
+  proxyConfig:
+    description: |
+      Централизованная настройка [ProxyConfig](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#ProxyConfig).
+    properties:
+      holdApplicationUntilProxyStarts:
+        description: |
+          Этот параметр заставляет sidecar-injector вставлять сайдкар на первое место в списке контейнеров пода. Также он добавляет postStart-хук дабы убедиться, что Envoy инициализировался раньше самого приложения. Это позволит избежать сетевых ошибок при старте пода.
```
Comment (konstantin-axenov) [inline ee/modules/110-istio/openapi/doc-ru-config-values.yaml:144]:
```suggestion
          Этот параметр заставляет sidecar-injector вставлять сайдкар на первое место в списке контейнеров пода. Также он добавляет postStart-хук, чтобы убедиться, что Envoy инициализировался раньше самого приложения. Это позволит избежать сетевых ошибок при старте пода.
```

File: werf.yaml [hunk 3]
```diff
@@ -333,7 +333,7 @@ ansible:
   - name: "Install GolangCI linter"
     shell: |
       curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
-        | sh -s -- -b $(go env GOPATH)/bin v1.46.2
+        | sh -s -- -b $(go env GOPATH)/bin v1.42.0
```
Comment (konstantin-axenov) [inline werf.yaml:336]: Wtf?

Comment (apolovov) [reply werf.yaml:336]:
```suggestion
        | sh -s -- -b $(go env GOPATH)/bin v1.46.2
```

**Review by yalosev — APPROVED** [review]

**Review by konstantin-axenov — APPROVED** [review]

**Pull request merged**